src/object/cli_obj.c

```diff
@@ -659,6 +659,8 @@ obj_reasb_req_fini(struct obj_reasb_req *reasb_req, uint32_t iod_nr)
 
 	for (i = 0; i < iod_nr; i++) {
 		iod = reasb_req->orr_iods + i;
+		if (iod == NULL)
+			return;
```

Comment:
continue?

You can also remove the L664 entirely.

Reply:
return -> continue or break so it can execute the two lines outside the for loop? I assume reasb_req->orr_iods does not have gaps in between orr_iods, so the 1st NULL iod found, no need to continue to check the next?

Are you sure to remove this?
                if (iod->iod_recxs != NULL)
                        D_FREE(iod->iod_recxs);
Who will free iod->iod_recxs ?

Reply:
I don't know, it's just odd to use a return in a for loop.  You'd have to find someone familiar with this code to see what it's supposed to be doing because it does look odd.

We don't as a rule check for NULL before calling D_FREE() was my point about L664

Reply:
@liuxuezhao seems like the owner of this function, could you please comment on this?